distributed/scheduler.py

```diff
@@ -2328,6 +2335,44 @@ def decide_worker(self, ts: TaskState) -> WorkerState:
             ts.state = "no-worker"
             return ws
 
+        total_nthreads: Py_ssize_t
+        if valid_workers is None:
+            total_nthreads = self._total_nthreads
+        else:
+            total_nthreads = 0
+            for ws in valid_workers:
+                total_nthreads += ws._nthreads
+
+        group_tasks_per_thread: double = (
+            (len(group) / total_nthreads) if total_nthreads > 0 else 0
+        )
+        if group_tasks_per_thread > 2 and sum(map(len, group._dependencies)) < 5:
```

Comment:
In https://github.com/dask/distributed/commit/28c5e4189894f17027025064a92c55458edaec72 I move all of the prelude here within the conditional.  I think it does a good job of isolating the code complexity of this special case at the cost of not handling some special cases (which I'm ok with I think?)